src/Reducer/Action.type.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  DECREMENT,
+  INCREMENT,
+  RESET,
+  SET_QUIZ,
+  UPDATE_ATTEMPT,
+  UPDATE_HIGHSCORE,
+  NEXT_QUESTION,
+  CLEAR_STATS,
+} from "./typeValues";
+
+export type Action =
+  | {
+      type: typeof SET_QUIZ;
```

Comment:
I don't think this is the safest type you can write (correct me if I'm wrong) but when you write 

`type: typeof SET_QUIZ`

and the type of SET_QUIZ is a string, then the type variable would accept any string. Rather you can assign it a string literal directly, like 

`type: "SET_QUIZ"`

Then you won't even need the typeValues file.

Reply:
Since SET_QUIZ is a constant variable, its type = "SET_QUIZ" according to typescript and not "string". Hence both the apporaches would be the same.